src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue

```diff
@@ -812,26 +784,54 @@ export default {
 						this.promoApplied = true;
 						this.promoErrorMessage = null;
 					}
-					this.$nextTick(() => {
-						this.loadingPromotion = false;
-					});
 				} else if (this.isMatchingCampaign) {
 					this.promoApplied = true;
-					this.loadingPromotion = false;
 				} else {
 					// Handle response and any potential errors
 					// > this reveals and prior error messages from the promo application
 					this.promoApplied = false;
-					this.loadingPromotion = false;
 				}
 
-				// Initialize loan query + observer there are no loans in the basket
+				this.loadingPromotion = false;
+
+				this.setInitialFilters();
+
 				this.showLoans = true;
 
 				this.setAuthStatus(this.kvAuth0?.user);
+
 				this.updateBasketState();
 			});
 		},
+		async setInitialFilters() {
```

Comment:
Is there something async in here?

Reply:
oh not any longer. I previous was putting `this.showLoans` above in a then block from setInitialFilters... turns out it wasn't needed in my testing.